Show notice in checkout with only Express Payments available

ralucaStan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes https://github.com/woocommerce/woocommerce/issues/58585
Closes https://linear.app/a8c/issue/WOOPLUG-4524/remove-error-message-on-blocks-checkout-when-only-express-payment

 It also fixes an unrelated additional fields blocks E2E test which was randomly failing. See commit https://github.com/woocommerce/woocommerce/pull/59075/commits/d4fc63b0858acb09ce8368c1fc8d96130bf1fd4c

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|![68747470733a2f2f75706c6f6164732e6c696e6561722e6170702f66363239666364652d366639622d343865382d386166332d3630656564383832336266662f32356662393233362d656330322d343661332d613363642d3635393762376662343437332f38396137326466652d393662612d346332332d](https://github.com/user-attachments/assets/e583489d-d976-4355-bd18-1acbb3be6261)   |    <img width="932" alt="Screenshot 2025-06-23 at 13 43 55" src="https://github.com/user-attachments/assets/e5a4ce78-1bbb-4672-af90-cf602f38b13d" /> |


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

On your local site make sure you have WooCommerce Subscription enabled, with a subscription product and no payment methods installed
### Set up an express payment method that accepts subs
1. Clone https://github.com/woocommerce/woocommerce-gateway-dummy
2. Edit `resources/js/frontend/index.js` and replace the code with the code below in order to only register an Express payments method
3. [Install the project ](https://github.com/woocommerce/woocommerce-gateway-dummy/blob/a94c4766e64c11df5ea089b13b7a5cfadc577a39/README.md)and activate the Dummy Gateway plugin on your test site

1. Add a subs to cart and go to checkout
2. Notice the info notice (blue notice) in the payments options area and the Express button from the plugin
3. Disable the Dummy Gateway 
4. Notice the error notice in the payments options area with the updated text

```
import { sprintf, __ } from "@wordpress/i18n";
import {
  registerPaymentMethod,
  registerExpressPaymentMethod,
} from "@woocommerce/blocks-registry";
import { decodeEntities } from "@wordpress/html-entities";
import { getSetting } from "@woocommerce/settings";

const settings = getSetting("dummy_data", {});

const defaultLabel = __("Dummy Payments", "woo-gutenberg-products-block");

const label = decodeEntities(settings.title) || defaultLabel;
/**
 * Content component
 */
const Content = () => {
  return decodeEntities(settings.description || "");
};
/**
 * Label component
 *
 * @param {*} props Props from payment API.
 */
const Label = (props) => {
  const { PaymentMethodLabel } = props.components;
  return <PaymentMethodLabel text={label} />;
};

const DummyExpress = {
  name: "dummy-express",
  label: <Label />,
  content: <button>Dummy Express</button>,
  edit: <button>Dummy Express</button>,
  canMakePayment: () => true,
  ariaLabel: label,
  supports: {
    features: ["products", "subscriptions"],
  },
};

registerExpressPaymentMethod(DummyExpress);

```

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ralucaStan committed:
Add notice for checkout with only express payments available

ralucaStan committed:
Add changelog

ralucaStan committed:
Add unit test

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

straku reviewed (approved):
Works as advertised. The only bit I am wondering about is the wording used in the error state:
> Only express payment methods are available. Please contact us for help placing your order.

I know contact is necessary when there's no payment methods, but with express one available shouldn't we suggest using it instead?

ralucaStan committed:
Update notice text

straku commented on plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/only-express-payments/index.tsx:
```diff
@@ -0,0 +1,30 @@
+			status="info"
+		>
+			{ __(
+				'Only express payment methods are available for this order. Please choose one to place your order.',
```
```suggestion
				'Only express payment methods are available for this order. Please select one to continue.',
```
I think the repeated "order" breaks my brain a little. What do you think about this suggestion?

ralucaStan committed:
Update plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/only-express-payments/index.tsx

Co-authored-by: Łukasz Strączyński <8038404+straku@users.noreply.github.com>

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

ralucaStan committed:
Improve test

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

ralucaStan committed:
Adding products to cart in tests instead in beforeEach
When adding in beforeEach the checkout wouldn't be loaded, resulting in the product not being added

ralucaStan committed:
Fix lint error

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

# Conflicts:
#	plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/payment-methods.tsx

ralucaStan committed:
Merge branch 'trunk' into wooplug-4524-remove-error-message-on-blocks-checkout-when-only-express

ralucaStan commented:
For visibility, I've also fixed another unrelated blocks E2E test failing on this PR in this commit https://github.com/woocommerce/woocommerce/pull/59075/commits/d4fc63b0858acb09ce8368c1fc8d96130bf1fd4c.
